huggingface_auth.py

```diff
@@ -0,0 +1,132 @@
+import asyncio
+import base64
+import os
+from datetime import datetime, timedelta
+from getpass import getpass
+
+import requests
+from huggingface_hub import HfApi
+
+from hivemind.proto.auth_pb2 import AccessToken
+from hivemind.utils.auth import TokenAuthorizerBase
+from hivemind.utils.crypto import RSAPublicKey
+from hivemind.utils.logging import get_logger
+
+
+logger = get_logger(__name__)
+
+
+class HuggingFaceAuthorizer(TokenAuthorizerBase):
+    _AUTH_SERVER_URL = 'https://collaborative-training-auth.huggingface.co'
+
+    def __init__(self, experiment_id: int, username: str, password: str):
+        super().__init__()
+
+        self._experiment_id = experiment_id
+        self._username = username
+        self._password = password
+
+        self._authority_public_key = None
+        self._hf_api = HfApi()
+
+    async def get_token(self) -> AccessToken:
+        token = self._hf_api.login(self._username, self._password)
```

Comment:
Here is what I would recommend as "latest" time to ask for the password (if it is possible)
```suggestion
        username = input("Huggingface username: ")
        password = getpass()
        token = self._hf_api.login(username, password)
```